### PR TITLE
lnt_sd_98--linkGroup method gives unexpected behaviour

### PR DESCRIPTION
**Pull request by aditya-7**
@JensEggers  @cordellcalitz 
linkGroup security API shows unexpected behaviour;
Find the test case in commit https://github.com/happner/happner/commit/c2d54e245f9ff6063671c6922248d175f617b455 failing.

### CODE REVIEW

File: test/b1-advanced-security.js [hunk 1]
```diff
@@ -347,9 +346,106 @@ describe('b1 - advanced security', function (done) {
...
+    this.timeout(2000);
+    var testUser = {
+      username : 'user1',
+      password : 'password',
+      custom_data : {
+        role : 'TEST GROUP ADMIN'
+      }
+    };
+
+    adminClient.exchange.security.addGroup(testGroupAdmin, function (e, result) {
```
Comment (JohanvdWest) [inline test/b1-advanced-security.js]: Please use the promise returned for these functions to avoid this deep nesting.

Comment (JohanvdWest) [inline test/b1-advanced-security.js]:
``` javascript
adminClient.exchange.security.addGroup(testGroupAdmin)
  .then(function() {
    return adminClient.exchange.security.addGroup(testGroupUser);
  })
  .then(function() {
    return adminClient.exchange.security.listGroups('*')
  })
  .then(function(groups) {
    for(var i=0;i<groups.length;i++){
       ...
    }
  })
  ...
  .then(done)
  .catch(done);
```

Comment (aditya-7) [inline test/b1-advanced-security.js]: @JohanvdWest @JensEggers 26/08-ak : Fixed in commit https://github.com/happner/happner/pull/188/commits/4dba96dc5d6f03cb135dac468534fc432684155d

**Comment by JohanvdWest** [comment]
OK, I am happy. Jens, you can merge if Simon is OK.

**Pull request merged**